include/crab/analysis/inter/top_down_inter_analyzer.hpp

```diff
@@ -1066,8 +1164,19 @@ class top_down_inter_transformer final
       // If we do not analyze precisely recursive functions then we
       // must start the analysis of a recursive procedure without
       // propagating from caller to callee (i.e., top).
-      domains::callsite_info<variable_t> callsite {cs.get_func_name(), cs.get_args(), cs.get_lhs(),
-						   fdecl.get_inputs(), fdecl.get_outputs()};
+      std::vector<std::vector<variable_t>> formal_cls, actual_cls;
+      group_regions_by_dsa_intrinsics(callee_cfg, cs.get_args(),
```

Comment:
This should be computed only once. We should compute it before the whole analysis starts, cache it and then just use it at each call. It should be cached in `m_ctx` similar to `get_widening_set()`.